dev/docker/web/epicast/README.md

```diff
@@ -0,0 +1,24 @@
+# `delphi_web_epicast`
+
+This image starts with Delphi's web server and adds the sources necessary for
+hosting the Epicast website. It further extends Delphi's web server by:
+
+- enabling the `mod_rewrite` extension
+- enabling the `mod_headers` extension
+- creating an empty `htpasswd` file
+
+This image includes the file
+[`settings.php`](assets/settings.php), which points to a local
+container running the
+[`delphi_database_epicast` image](../../database/epicast/README.md).
+
+To start a container from this image, run:
+
+```bash
+docker run --rm -p 10080:80 \
+  --network delphi-net --name delphi_web_epicast \
+  delphi_web_epicast
+```
+
+You should be able to login and interact with the website (e.g. submitting
+predictions) by visiting `http://localhost:10080/` in a web browser.
```

Comment:
Is there a way to make this work for a pandemic setup where, e.g., my development machine is on campus and I'm shelling into it from home? Are we talking X forwarding or is there a way to expose this address outside localhost?

Reply:
oh interesting, that's a good use-case. no need to X-forward. you should be able to use a reverse ssh tunnel to forward localhost:10080 on your at-home machine to localhost:10080 on your at-work machine.

from your at-home machine, this command should work:
```
ssh -N -R localhost:10080:localhost:10080 your-username@at-work-machine
```

This works by listening on port 10080 at-home and tunneling connections to port 10080 at-work. (The first `localhost:10080` is where to listen at-home, the second `localhost:10080` is where to connect at-work.)

This is probably something that should go in the guide. Do you mind letting me know if this works for you?

Reply:
(and to be clear, after the `ssh` command above, you can then open `http://localhost:10080/` on your at-home machine, and you should see your "local" (at-work) instance of the website.)

Reply:
I have the db and web server running in docker on my work machine, and I can load the site in lynx from there. Alas, the ssh tunnel fails:

```
$ ssh -N -R localhost:10080:localhost:10080 <me>@<work>
<me>@<work>'s password: 
Warning: remote port forwarding failed for listen port 10080
^C$ ssh -N -R 127.0.0.1:10080:127.0.0.1:10080 <me>@<work>
<me>@<work>'s password: 
Warning: remote port forwarding failed for listen port 10080
```

--which seems weird; I'm not a tunneling pro but I've done something like this before. I'll have more time to debug tomorrow morning.

Reply:
ahh my bad. this should be a _forward_ tunnel, not reverse. just need to swap out `-R` with `-L` as in:

```
ssh -N -L localhost:10080:localhost:10080 your-username@at-work-machine
```